exonum/src/blockchain/service.rs

```diff
@@ -172,6 +172,10 @@ pub trait Service: Send + Sync + 'static {
         Value::Null
     }
 
+    /// A service execution. This method is invoked for each service after execution
+    /// of all transactions in the mempool but before `handle_commit` handler.
```

Comment:
- I think an example is needed to explain possible use cases.
- Please document the order in which `execute` are invoked for services. (Ideally, there should be a couple of tests checking this order.)
- "execution of all transactions in the mempool" - do you mean "...in a block"?

Reply:
@slowli I don't understand why the invoke order is important for us here. We expect that each service will work with own indices and own data, don't we? And a modifying indices from other services it's a bad practice ? At least now.

Reply:
I think that you should add comment about the unspecified `Service:execute` order.